src/legacy/ui/public/index_patterns/index_patterns.ts

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// @ts-ignore
+import { fieldFormats } from '../registry/field_formats';
+
+import { IndexPattern } from './_index_pattern';
+import { createIndexPatternCache } from './_pattern_cache';
+import { IndexPatternsApiClient } from './index_patterns_api_client';
+import { SavedObjectsClient, SimpleSavedObject } from '../saved_objects';
+import { UiSettingsClient } from '../../../../core/public';
+
+const indexPatternCache = createIndexPatternCache();
+const apiClient = new IndexPatternsApiClient();
+
+export class IndexPatterns {
+  fieldFormats: fieldFormats;
+
+  private config: UiSettingsClient;
+  private savedObjectsClient: SavedObjectsClient;
+  private savedObjectsCache?: Array<SimpleSavedObject<{}>> | null;
+
+  constructor(config: UiSettingsClient, savedObjectsClient: SavedObjectsClient) {
+    this.config = config;
+    this.savedObjectsClient = savedObjectsClient;
+  }
+
+  private async loadSavedObjects() {
+    this.savedObjectsCache = (await this.savedObjectsClient.find({
+      type: 'index-pattern',
+      fields: [],
+      perPage: 10000,
+    })).savedObjects;
```

Comment:
Yeah, that's a good question. Saved objects API has a bunch of [error helpers](https://github.com/elastic/kibana/blob/master/src/core/server/saved_objects/service/lib/errors.ts), so I suppose this could through an error, though AFAICT we weren't previously handling these errors anyway. So I'd recommend we address this separately (and perhaps more broadly) across index patterns in a separate PR.